### PR TITLE
Rspec model borrowed_request

### PR DESCRIPTION
**Pull request by khanhpn-2944**
WHAT
- Create rspec and add some factories, model tests

HOW
- Setup rspec (gem and install)
- Add some related fatories
- Add unit tests for model borrowed_request

Evidence (Screenshot or Video)
Screenshot from 2022-04-19 17-44-56

### CODE REVIEW

**Comment by khanhpn-2944** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -19,12 +19,16 @@ gem "rails", "~> 6.1.4", ">= 6.1.4.1"
 gem "rails-i18n"
 gem "rake", "13.0.6"
 gem "sass-rails", ">= 6"
+gem "simplecov"
+gem "simplecov-rcov"
```
Comment (vanvtt-0952) [inline Gemfile]: đưa vào group test nhé

File: spec/factories/authors.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :author do
+    name {Faker::Name.name}
+    detail {"This is an author."}
```
Comment (vanvtt-0952) [inline spec/factories/authors.rb]: Dùng faker gem luôn đi em ơi

File: spec/models/borrowed_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+      it "should returned" do
+        request.update status: 2
+        expect(request.returned?).to eq true
+      end
+
+      it "should rejected" do
+        request.update status: 3
+        expect(request.rejected?).to eq true
+      end
```
Comment (vanvtt-0952) [inline spec/models/borrowed_request_spec.rb]: tesst enum chỉ cần 
`it { should define_enum_for(:status).
            with(:.... tự điền giá trị em định nghĩa là dc) }`

**Comment by khanhpn-2944** [comment]
ready

**Pull request merged**